From 209aa62950e503494716fd046f8f5cb546bf57d4 Mon Sep 17 00:00:00 2001
From: Jonathan Kang <jonathankang@gnome.org>
Date: Wed, 4 Sep 2024 15:16:25 +0800
Subject: [PATCH] zypp: Mark the correct packages to-be-installed

The package list passed to backend_update_packages_thread() is already
the packages that are about to be installed. It doesn't make sense to
mark the update candidate of those packages to be installed.

https://bugzilla.suse.com/show_bug.cgi?id=1227389
---
 backends/zypp/pk-backend-zypp.cpp | 5 -----
 1 file changed, 5 deletions(-)

diff --git a/backends/zypp/pk-backend-zypp.cpp b/backends/zypp/pk-backend-zypp.cpp
index dd3f19ee0..2d4a51238 100644
--- a/backends/zypp/pk-backend-zypp.cpp
+++ b/backends/zypp/pk-backend-zypp.cpp
@@ -3524,11 +3524,6 @@ backend_update_packages_thread (PkBackendJob *job, GVariant *params, gpointer us
 				zypp_backend_finished_error (job, PK_ERROR_ENUM_DEP_RESOLUTION_FAILED, "Package %s is not installed", package_ids[i]);
 				return;
 			}
-			item = sel->updateCandidateObj();
-			if (!item) {
-				 zypp_backend_finished_error(job, PK_ERROR_ENUM_DEP_RESOLUTION_FAILED, "There is no update candidate for %s", sel->installedObj().satSolvable().asString().c_str());
-				return;
-			}
 		}
 
 		item.status ().setToBeInstalled (ResStatus::USER);
-- 
2.46.0

